.github/workflows/test.yml

```diff
@@ -0,0 +1,16 @@
+name: test
```

Comment:
What's the benefit of declaring a new workflow via GH instead of using the circleCI one ?
The docker image push could be written in the circleCI workflow IMO, I think it would be less confusing to have a single place for automated workflows (or feel free to move it all on GH ^^, however, notice that the circle wf automatically publishes the compiled binary to tagged releases).